saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/Saml2AuthenticationRequestContext.java

```diff
@@ -39,15 +40,25 @@
 
 	private final String relayState;
 
+	private final RequestedAuthnContext requestAuthnContext;
+
+	private final boolean isPassive;
+
+	private final boolean forceAuth;
+
 	protected Saml2AuthenticationRequestContext(RelyingPartyRegistration relyingPartyRegistration, String issuer,
-			String assertionConsumerServiceUrl, String relayState) {
+			String assertionConsumerServiceUrl, String relayState, RequestedAuthnContext requestAuthnContext,
```

Comment:
Since this is a protected method, we can't change the signature; otherwise, it will break folks who upgrade. Please add a new constructor, leaving this constructor as-is.

I'd recommend the new constructor be `private` and have it take an instance of the `Builder`.